### PR TITLE
#153 Added perf-rec tests for the views

### LINKED ISSUE #153: count/record queries on every API view
using [django-perf-rec](https://pypi.python.org/pypi/django-perf-rec), for example, we could have metrics on each API view, and eventually improve them using DRF tools to remove duplicate queries, use joins where needed, etc.

### PR DESCRIPTION
**Pull request by alexdashkov**
## Review

This PR closes #153 

Please check the records structure. 

* [x] Tests
* [x] `CHANGELOG.rst` Updated
* [ ] Delete your branch

### CODE REVIEW

File: demo/tests/test_perfs_rec.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+from django_perf_rec import TestCaseMixin
+
+
+class TestCase(TestCaseMixin, OrigTestCase):
+    """
+    Add use perf rec mixin
+    """
+
+    c = Client()
```
Comment (wo0dyn) [inline demo/tests/test_perfs_rec.py]: You don't need to instantiate `Client` here, it's already accessible via `self.client` in a [Django TestCase](https://docs.djangoproject.com/en/1.8/topics/testing/tools/#default-test-client).

File: demo/tests/test_perfs_rec.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    yml_dir = 'perfs'
+    file_name_pattern = 'django%s-py%s.yml'
+
+    def get_yml_dir_path(self):
+        """
+        Function to generate a path for all the Python and Django version
```
Comment (wo0dyn) [inline demo/tests/test_perfs_rec.py]: You can remore `Function to` here.

```python
def get_yml_dir_path(self):
    """
    Generate a path for all the Python and Django version
    """
```

File: demo/tests/test_perfs_rec.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            record_name=record_name
+        )
+
+
+class TestPerfRec(TestCase):
+    """
+    """
+    fixtures_folder = 'fixtures'
+
+    def setUp(self):
```
Comment (wo0dyn) [inline demo/tests/test_perfs_rec.py]: You can use `setUpClass()` classmethod, to load files only once. See warning section on https://docs.djangoproject.com/en/1.8/topics/testing/tools/#simpletestcase

File: demo/tests/test_perfs_rec.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        )
+        self.form_data_changed = json.load(
+            open(os.path.join(fixtures_path, 'form-data-changed.json'))
+        )
+
+    def test_access_list_perf_rec(self):
+
+        with self.record_performance(record_name='accesses-list'):
+            url = reverse('formidable:accesses_list')
+            self.c.get(url)
```
Comment (wo0dyn) [inline demo/tests/test_perfs_rec.py]: `self.client.get(url)`

**Comment by alexdashkov** [comment]
@wo0dyn Thank you for your feedback, I fixed PR.

File: demo/tests/test_perfs_rec.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            path = self.get_yml_dir_path()
+
+        return super(TestCase, self).record_performance(
+            path=path,
+            record_name=record_name
+        )
+
+
+class TestPerfRec(TestCase):
+    """
```
Comment (brunobord) [inline demo/tests/test_perfs_rec.py]: empty docstrings aren't really useful here.

**Comment by alexdashkov** [comment]
@brunobord, thank you for your feedback. I made the changes. I notice a weird thing that before I had the different files for the different Django version(especially with the session table call) but now it's not the case now.

**Comment by brunobord** [comment]
too bad you merged master into this branch, instead of trying to rebase it. but, well... that won't make any difference in the end.

**Comment by wo0dyn** [comment]
Agree w/ @brunobord. Please @alexdashkov rebase against `master` instead of merging. It'll make the history way cleaner. 😉

**Comment by alexdashkov** [comment]
@brunobord I already know this, @wo0dyn wrote me about this on the first PR. It was just a test, what does GitHub do when it proposes to update the base branch. I'll remove this commit.

**Comment by brunobord** [comment]
cool! thx a bunch

**Comment by alexdashkov** [comment]
You are welcome

**Pull request merged**